pkg/discovery/network/canal.go

```
@@ -52,14 +52,10 @@ func discoverCanalFlannelNetwork(clientSet kubernetes.Interface) (*ClusterNetwor
 		PodCIDRs:      []string{*podCIDR},
 	}
 
-	// Try to networkPluginsDiscovery the service CIDRs using the generic functions
-	genNetwork, err := discoverGenericNetwork(clientSet)
-	if err != nil {
-		return nil, err
-	}
-
-	if genNetwork != nil {
-		clusterNetwork.ServiceCIDRs = genNetwork.ServiceCIDRs
+	// Try to detect the service CIDRs using the generic functions
+	clusterIPRange, err := findClusterIPRange(clientSet)
+	if err == nil && clusterIPRange != "" {
```

As commented elsewhere, I think we should propagate an error.